eng/scripts/Set-VcpkgWriteModeCache.ps1

```diff
@@ -0,0 +1,41 @@
+param(
+    [string] $StorageAccountKey
+)
+
+
+."$PSScriptRoot/../common/scripts/Helpers/PSModule-Helpers.ps1"
+
+Write-Host "`$env:PSModulePath = $($env:PSModulePath)"
+
+# Work around double backslash
+if ($IsWindows) {
+    $hostedAgentModulePath = $env:SystemDrive + "\\Modules"
+    $moduleSeperator = ";"
+} else {
+    $hostedAgentModulePath = "/usr/share"
+    $moduleSeperator = ":"
+}
+$modulePaths = $env:PSModulePath -split $moduleSeperator
+$modulePaths = $modulePaths.Where({ !$_.StartsWith($hostedAgentModulePath) })
+$AzModuleCachePath = (Get-ChildItem "$hostedAgentModulePath/az_*" -Attributes Directory) -join $moduleSeperator
```

Comment:
Was all this copied out of the helper code?

Reply:
It was. There's a problem in `PSModule-Helpers.ps1` https://github.com/Azure/azure-sdk-tools/issues/3039

Now that I'm thinking about it more, this file also exists in C++... we could put this file into eng/common.

Reply:
Can we fix the underlying issue in tools instead of copying all this code?